client/src/main/App.tsx

```diff
@@ -277,52 +135,21 @@ class App extends React.Component<AppProps, AppState> {
       snackBarOpen,
       snackBarSeverity,
       snackBarText,
-      user,
-      projects,
-      tasks,
       appTheme,
+      projectIds,
     } = this.state;
-    const {
-      handleSnackBarClose,
-      alert,
-      updateSettings,
-      sendUpdatedUserToServer,
-      setProjects,
-      setProject,
-      setUser,
-      setTask,
-      setTasks,
-      setTheme,
-    } = this;
+    const { handleSnackBarClose, alert, setTheme } = this;
     return (
       <div className="App">
         <ThemeProvider theme={appTheme}>
           <TopMenuBar
             githubClientId={githubClientId}
-            baseServerUrl={baseServerUrl}
-            sendUpdatedUserToServer={sendUpdatedUserToServer}
             alert={alert}
-            userSettings={user ? user.settings : undefined}
-            updateSettings={updateSettings}
             appTheme={appTheme}
             setTheme={setTheme}
-            loggedIn={!!user}
+            loggedIn={!!ClientData.getUser()}
```

Comment:
I believe the TopMenuBar is not getting 'notified' of the change in login state from ClientData.

Reply:
Good catch. Fixing that now. Thank you

Reply:
Glad you are looking into it, trying to fix this without fully understanding the ClientData changes was making me go around in circles.

Reply:
To fix it, it looks like adding the capability to add listeners to the user object itself will be a clean option. Then the TopMenuBar will be a listener of changes to that and update it's `loggedIn` property accordingly. Going to try that now.